### ISSUE TITLE
documentation on tree design

### ISSUE THREAD

**Issue by balmas** [latin-language-toolkit/arethusa#689]
@Christof and @LFDM - can either of you provide a brief summary of the design decisions and issues that were involved in the design of the tree display, particularly around the embedding of angular directives within the SVG? I know there was lots of back and forth and discussion about this, and as @PonteIneptique begins to work on alternate graph visualizations in Arethusa it would be helpful to have access to some of the background here.

**Comment by PonteIneptique** [comment]
Hey guys,
I'd love if you could have a look at my code here : https://github.com/ponteineptique/arethusa/tree/network-d3 . The span are "compiled" after they are appended, making it impossible to adapt foreignObject height to their height because at the time I do .childNode and .offsetWidth, the element inside the main token object do not exist. It worked when I - wrongly - added a compile + string on token.js (see the code before I removed it https://github.com/PonteIneptique/arethusa/blob/a364306ac46758650654cd0ede16d5b1c430d0bf/app/js/arethusa.core/directives/token.js )

**Comment by PonteIneptique** [comment]
Guys, I found the answer in your code and dagre. I love you and you're so clever : insertPlaceholder before directive to make foreignObject has the right size. That's cool.

**Comment by balmas** [comment]
@PonteIneptique for posterity let's include a code ref. Are these the right bits? 

https://github.com/latin-language-toolkit/arethusa/blob/master/app/js/arethusa.core/tree.js#L139-L156
https://github.com/latin-language-toolkit/arethusa/blob/master/app/js/arethusa.core/tree.js#L344-L350
https://github.com/latin-language-toolkit/arethusa/blob/master/app/js/arethusa.core/tree.js#L452-L461

**Comment by balmas** [comment]
Or even better, could you add a ngdoc comment to the tree.js that describes the process?

**Comment by PonteIneptique** [comment]
I think that would be a mistake for now, as the tree.js does not reflect the whole process (a lot of things are taken care of in Dagre-D3) But when open_data_graph is available, I'll do the ngdoc !

**Comment by balmas** [comment]
Actually I think it really belongs in the (currently empty) core guide, rather than as documentation on a specific plugin or module. https://github.com/latin-language-toolkit/arethusa/tree/master/docs/custom/ngdoc/core_guide

**Comment by PonteIneptique** [comment]
Yes and no : while there is a lot of things which could be documented here, I think this specific part is too tied to Dagre (Dagre takes care of a lot, including resizing. The new graph took from dagre's code). Another thing through, I think either the new graph should be moved to core, or tree should be move out of core.

**Comment by balmas** [comment]
Well, wherever we put it, documentation on how to use dagre to create a graph-based plugin display in an Arethusa plugin would be a good thing to have.

As for whether the new graph belongs in core or tree doesn't, maybe they both really belong in arethusa.util ? or a new module, arethusa.viz ?
